pkg/resources/core/v1/secret/mutator.go

```diff
@@ -75,7 +97,7 @@ func (m *Mutator) MutatingWebhook(clientConfig admissionregistrationv1.WebhookCl
 	mutatingWebhook.MatchConditions = []admissionregistrationv1.MatchCondition{
 		{
 			Name:       "filter-by-secret-type-cloud-credential",
-			Expression: `request.operation == 'DELETE' || (object != null && object.type == "provisioning.cattle.io/cloud-credential")`,
+			Expression: `request.operation == 'DELETE' || (object != null && object.type == "provisioning.cattle.io/cloud-credential" && request.operation == 'CREATE') || (object != null && object.metadata.namespace == "` + localUserPasswordsNamespace + `")`,
```

Comment:
Maybe some line breaks to make this a little more readable?
```suggestion
			Expression: `request.operation == 'DELETE' ||
			(object != null && object.type == "provisioning.cattle.io/cloud-credential" && request.operation == 'CREATE') || 
			(object != null && object.metadata.namespace == "` + localUserPasswordsNamespace + `")`,
```